CRM/Member/Form/MembershipRenewal.php

```diff
@@ -598,6 +598,10 @@ public function postProcess() {
 
       $payment = CRM_Core_Payment::singleton($this->_mode, $this->_paymentProcessor, $this);
 
+      if ($paymentParams['auto_renew']) {
```

Comment:
This will throw a notice if you don't check 'Membership renewed automatically', should be isset($paymentParams['auto_renew']).